src/main/java/com/thevoxelbox/voxelsniper/config/VoxelSniperConfigLoader.java

```diff
@@ -28,22 +34,100 @@
     private static final String DEFAULT_CYLINDER_CENTER = "default-cylinder-center";
     protected static final String BRUSH_PROPERTIES = "brush-properties";
 
+    private static final int CONFIG_VERSION_VALUE = 1;
     private static final boolean DEFAULT_MESSAGE_ON_LOGIN_ENABLED = false;
     private static final BlockType DEFAULT_BLOCK_MATERIAL_VALUE = BlockTypes.AIR;
     private static final BlockType DEFAULT_REPLACE_BLOCK_MATERIAL_VALUE = BlockTypes.AIR;
     private static final int DEFAULT_BRUSH_SIZE_VALUE = 3;
     private static final int DEFAULT_LITESNIPER_MAX_BRUSH_SIZE = 5;
+    private static final List<BlockType> DEFAULT_LITESNIPER_RESTRICTED_MATERIALS = Arrays.asList(
+            BlockTypes.BARRIER,
+            BlockTypes.BEDROCK
+    );
     private static final int DEFAULT_BRUSH_SIZE_WARNING_THRESHOLD = 20;
     private static final int DEFAULT_VOXEL_HEIGHT_VALUE = 1;
     private static final int DEFAULT_CYLINDER_CENTER_VALUE = 0;
+    private static final Map<String, Map<String, Object>> DEFAULT_BRUSH_PROPERTIES = null;
 
+    private final VoxelSniperPlugin plugin;
     private final FileConfiguration config;
 
     /**
-     * @param config Configuration that is going to be used.
+     * Create a new cached voxel configuration loader.
+     *
+     * @param plugin plugin instance
+     * @param config configuration that is going to be used.
      */
-    public VoxelSniperConfigLoader(FileConfiguration config) {
+    public VoxelSniperConfigLoader(VoxelSniperPlugin plugin, FileConfiguration config) {
+        this.plugin = plugin;
         this.config = config;
+        updateConfig();
+    }
+
+    /**
+     * Update config settings the current config version and apply changes.
+     */
+    public void updateConfig() {
+        int currentConfigVersion = getConfigVersion();
+        if (currentConfigVersion != CONFIG_VERSION_VALUE) {
+            Bukkit.getLogger().warning("Invalid config file found! Trying to apply required changes...");
```

Comment:
`Bukkit#getLogger` returns the instance of the logger the server uses, not the one of the plugin.